daliuge-engine/dlg/graph_loader.py

```diff
@@ -417,8 +430,13 @@ def _createApp(dropSpec, dryRun=False, session_id=None):
         module = importlib.import_module(".".join(parts[:-1]))
         appType = getattr(module, parts[-1])
     except (ImportError, AttributeError, ValueError):
-        raise InvalidGraphException(
-            "drop %s specifies non-existent application: %s" % (oid, appName)
+        err = "drop %s specifies non-existent application: %s" % (oid, appName)
+        return PyFuncApp(
```

Comment:
Personally, I do not like this approach of using an AppDROP to do our error management. I think it makes much more sense to have an error manager that we use to do this, rather than relying on the the `PyFuncApp`. I think this approach needs further discussion to figure out how we want our error reporting to work moving forward.